src/clj/puppetlabs/services/master/master_service.clj

```diff
@@ -53,36 +80,31 @@
-     (let [route-config (core/get-master-route-config ::master-service config)
-           path (core/get-master-mount ::master-service route-config)
-           ring-app (core/construct-root-routes puppet-version
-                                                use-legacy-auth-conf
-                                                jruby-service
-                                                get-code-content
-                                                handle-request
-                                                (get-auth-handler)
-                                                environment-class-cache-enabled)
-           routes (comidi/context path ring-app)
-           comidi-handler (comidi/routes->handler routes)
-           ring-handler (comidi/wrap-with-route-metadata comidi-handler routes)]
-       ;; if the webrouting config uses the old-style config where
-       ;; there is a single key with a route-id, we need to deal with that
-       ;; for backward compat.  We have a hard-coded assumption that this route-id
-       ;; must be `master-routes`.  In Puppet Server 2.0, we also supported a
-       ;; key called `invalid-in-puppet-4` in the same route config, even though
-       ;; that key is no longer used for Puppet Server 2.1 and later.  We
-       ;; should be able to remove this hack as soon as we are able to get rid
-       ;; of the legacy routes.
-       (if (and (map? route-config)
-                (contains? route-config :master-routes))
-         (add-ring-handler this
-                           ring-handler
-                           {:route-id :master-routes
-                            :normalize-request-uri true})
-         (add-ring-handler this
-                           ring-handler
-                           {:normalize-request-uri true}))))
-   context)
+
+     ;; if the webrouting config uses the old-style config where
+     ;; there is a single key with a route-id, we need to deal with that
+     ;; for backward compat.  We have a hard-coded assumption that this route-id
+     ;; must be `master-routes`.  In Puppet Server 2.0, we also supported a
+     ;; key called `invalid-in-puppet-4` in the same route config, even though
+     ;; that key is no longer used for Puppet Server 2.1 and later.  We
+     ;; should be able to remove this hack as soon as we are able to get rid
+     ;; of the legacy routes.
+     (if (and (map? route-config)
+              (contains? route-config :master-routes))
+       (add-ring-handler this
+                         ring-handler
+                         {:route-id :master-routes
+                          :normalize-request-uri true})
+       (add-ring-handler this
+                         ring-handler
+                         {:normalize-request-uri true}))
+
+     (register-status
+      "master"
+      (status-core/get-artifact-version "puppetlabs" "puppetserver")
+      1
+      (partial core/v1-status metrics))
```

Comment:
This is an interesting idea, thanks, @rlinehan.  I think, though, that I'd prefer to not embed any PE-specific/aware code in OSS puppetserver.  I'd like to be able to control how/if we deprecate (or eventually remove) the registration of status info for the pe-master (as well as pe-jruby-metrics and pe-puppet-profiler) services via code that sits in pe-puppet-server-extensions.  Having to make a change in OSS Puppet Server and a release/rollup of that change into pe-puppet-server-extensions in order to remove the registration of the "pe-master" service name from status info later wouldn't feel right to me.

Given that it sounds like we're probably going to need to keep providing status info under both the PE and OSS names of these services for some period of time to preserve compatibility, I'm leaning toward having pe-puppetserver bootstrap both OSS and PE versions of the services during the transition period.  The OSS versions of the services would do most of the real work.  The PE services would basically just do a `register-status` call with the appropriate `pe-*` service names for compatibility, reusing the actual metrics objects / functions from the related OSS service implementations.  Then when/if we want to remove the PE services references, we can just delete the service implementations (and bootstrap config references) entirely from the PE codebase - on whatever cadence would most make sense for PE.

Would that approach make sense to you?